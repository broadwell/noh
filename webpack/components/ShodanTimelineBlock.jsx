import React from "react";
import PropTypes from "prop-types";

// Rather than passing in style by props, these should be
// calculated from duration of section and intensity
const ShodanTimelineBlock = props => {
  // should actually check to be sure that intensity is not greater than max
  const heightNum = props.intensity / props.maxIntensity * 100;
  return (
    <div
      className="shodan-map__item"
      style={{
        left: props.left,
        width: props.width,
        height: `${String(heightNum)}%`
      }}
      data-tooltip={props.name}
    />
  );
};

ShodanTimelineBlock.propTypes = {
  left: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  maxIntensity: PropTypes.number.isRequired,
  intensity: PropTypes.number.isRequired
};

export default ShodanTimelineBlock;
