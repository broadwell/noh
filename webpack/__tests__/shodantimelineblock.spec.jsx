import React from "react";
import { shallow } from "enzyme";
import ShodanTimelineBlock from "../components/ShodanTimelineBlock";

describe("<ShodanTimelineBlock>", () => {
  it("renders as expected", () => {
    const component = shallow(
      <ShodanTimelineBlock name="Kiri" left="8%" width="10%" height="90%" />
    );
    expect(component).toMatchSnapshot();
  });
});
