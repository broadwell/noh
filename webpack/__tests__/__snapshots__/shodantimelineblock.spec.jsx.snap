// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShodanTimelineBlock> renders as expected 1`] = `
<div
  className="shodan-map__item"
  data-tooltip="Kiri"
  style={
    Object {
      "height": "90%",
      "left": "8%",
      "width": "10%",
    }
  }
/>
`;
